packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -0,0 +1,394 @@
+    List<String> path, 
+    List<dynamic> aotSizeJson,
+  ) {
+    Map<String, dynamic> currentLevel = apkAnalysisJson;
+    while (path.isNotEmpty) {
+      final List<Map<String, dynamic>> children = currentLevel['children'] as List<Map<String, dynamic>>;
+      final Map<String, dynamic> childWithPathAsName = children.firstWhere(
+        (Map<String, dynamic> child) => child['n'] as String == path.first,
+      );
+      path.removeAt(0);
+      currentLevel = childWithPathAsName;
+    }
+    currentLevel['children'] = treemapFromJson(aotSizeJson)['children'];
+    return apkAnalysisJson;
+  }
+
+  /// A pretty printer for an entity with a size.
+  void _printEntitySize(
+    String entityName,
+    int numBytes,
+    int level, {
+    bool showColor = true,
+    }) {
+    final bool emphasis = level <= 1;
+    final String formattedSize = _prettyPrintBytes(numBytes);
+
+    TerminalColor color = TerminalColor.green;
+    if (formattedSize.endsWith('MB')) {
+      color = TerminalColor.cyan;
+    } else if (formattedSize.endsWith('KB')) {
+      color = TerminalColor.yellow;
+    }
+
+    final int spaceInBetween = tableWidth - level * 2 - entityName.length - formattedSize.length;
+    _logger.printStatus(
+      entityName + ' ' * spaceInBetween,
+      newline: false,
+      emphasis: emphasis,
+      indent: level * 2,
+    );
+    _logger.printStatus(formattedSize, color: showColor ? color : null);
+  }
+
+  String _prettyPrintBytes(int numBytes) {
+    const int kB = 1024;
+    const int mB = kB * 1024;
+    if (numBytes < kB) {
+      return '$numBytes B';
+    } else if (numBytes < mB) {
+      return '${(numBytes / kB).round()} KB';
+    } else {
+      return '${(numBytes / mB).round()} MB';
+    }
+  }
+}
+
+class SymbolNode {
+  SymbolNode(
+    this.name, {
+    int value = 0,
+  })  : assert(name != null),
+        assert(value != null),
+        _children = <String, SymbolNode>{},
+        _value = value;
+
+  /// The human friendly identifier for this node.
+  String name;
+
+  int _value;
+  int get value {
+    _value ??= children.fold(
+      0,
+      (int accumulator, SymbolNode node) => accumulator + node.value,
+    );
+    return _value;
+  }
+
+  void addValue(int valueToBeAdded) {
+    _value += valueToBeAdded;
+  }
+
+  SymbolNode get parent => _parent;
+  SymbolNode _parent;
+
+  Iterable<SymbolNode> get children => _children.values;
+  final Map<String, SymbolNode> _children;
+
+  SymbolNode childByName(String name) => _children[name];
+
+  SymbolNode addChild(SymbolNode child) {
+    assert(child.parent == null);
+    assert(!_children.containsKey(child.name),
+        'Cannot add duplicate child key ${child.name}');
+
+    child._parent = this;
+    _children[child.name] = child;
+    SymbolNode ancestor = this;
+    while (ancestor != null) {
+      ancestor._value += child.value;
+      ancestor = ancestor.parent;
+    }
+    return child;
+  }
+
+  List<SymbolNode> get ancestors {
```

Comment:
Neither ancestors nor siblings are used in the test, so they currently have 0 coverage